frontend/src/admin/components/dropdown-menu/dropdown-menu.tsx

```diff
@@ -0,0 +1,37 @@
+import React, {useState} from "react";
+import DropdownMenuIcon from "../dropdown-menu-icon/dropdown-menu-icon";
+import "./dropdown-menu.scss";
+
+interface DropdownMenuProps {
+    children: React.ReactNode
+}
+
+export default function DropdownMenu( {children}: DropdownMenuProps) {
+    const [showDropdown, setShowDropdown] = useState(false);
+
+    function toggleDropdown() {
+        setShowDropdown(!showDropdown);
+    }
+
+    function hideDropdown(event: React.FocusEvent<HTMLButtonElement>): void {
+        if (event.currentTarget === event.target) {
```

Comment:
A good thing for you to start practicing is a sort of "defensive programming" where instead of writing "if condtion is true then do thing" you would be writing "if condition not true then stop execution".
In this case it would look like this:
```
function hideDropdown(event: ...): void {
  if (event.currentTarget !== event.target) {
    return
  }
  setShowDropdown(false);
}
```
This is an easy way to avoid unnecessary nesting and uses the mindset that you assert that all your inputs/condition are in the state you want them to to be in before you start executing your logic.

Reply:
Super nice input! I did not know this term. It gives the code a more robust structure. Cool!